code/datums/extensions/abilities/abilities_mob.dm

```diff
@@ -2,10 +2,10 @@
 	var/datum/extension/abilities/abilities
 	if(create_if_missing)
 		abilities = get_or_create_extension(src, /datum/extension/abilities)
-	else
+	else if(has_extension(src, /datum/extension/abilities))
 		abilities = get_extension(src, /datum/extension/abilities)
-		if(!abilities)
-			return null
+	if(!abilities)
```

Comment:
this breaks `create_if_missing`. also you removed a lot of cases where that arg was passed in, is it deprecated now?

Reply:
right, i was looking at lines 10-11, since `add_ability_handler()` also does `get_or_create_extension()`. missed that it was checked earlier. looks good